From ec2cfb21aef91da73cb9c14926c5a4335f050988 Mon Sep 17 00:00:00 2001
From: "Hebbar, Gururaja" <gururaja.hebbar@ti.com>
Date: Thu, 3 May 2012 15:26:11 +0530
Subject: [PATCH 0952/1092] ARM: OMAP2+: am335x: Add MMC0 support for EVM-SK

This patch adds initial MMC0 support for EVM-SK board.
Since this Board has Micro-SD slot, write-protect detection is disabled.

Signed-off-by: Hebbar, Gururaja <gururaja.hebbar@ti.com>
---
 arch/arm/mach-omap2/board-am335xevm.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm/mach-omap2/board-am335xevm.c b/arch/arm/mach-omap2/board-am335xevm.c
index 1420ecc..f3a1f0a 100644
--- a/arch/arm/mach-omap2/board-am335xevm.c
+++ b/arch/arm/mach-omap2/board-am335xevm.c
@@ -1538,6 +1538,7 @@ static void mmc0_init(int evm_id, int profile)
 	switch (evm_id) {
 	case BEAGLE_BONE_A3:
 	case BEAGLE_BONE_OLD:
+	case EVM_SK:
 		setup_pin_mux(mmc0_common_pin_mux);
 		setup_pin_mux(mmc0_cd_only_pin_mux);
 		break;
@@ -1722,6 +1723,7 @@ static struct evm_dev_cfg beaglebone_dev_cfg[] = {
 
 /* EVM - Starter Kit */
 static struct evm_dev_cfg evm_sk_dev_cfg[] = {
+	{mmc0_init,	DEV_ON_BASEBOARD, PROFILE_ALL},
 	{NULL, 0, 0},
 };
 
@@ -1809,6 +1811,9 @@ static void setup_starterkit(void)
 {
 	pr_info("The board is a AM335x Starter Kit.\n");
 
+	/* Starter Kit has Micro-SD slot which doesn't have Write Protect pin */
+	am335x_mmc[0].gpio_wp = -EINVAL;
+
 	_configure_device(EVM_SK, evm_sk_dev_cfg, PROFILE_NONE);
 }
 
-- 
1.7.11.2

